### PR TITLE
Worksheet search

### PR DESCRIPTION
**Pull request by sagarkoshti1990**
## Description

Search Worksheets (Normal and Advance Search)

### Changes

 - New Search layout
 - Worksheet page

### CODE REVIEW

File: packages/common-lib/src/components/SearchLayout.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  minStringLenght,
+  notFoundMessage,
+  imageUrl,
+  children,
+  search,
+  setSearch,
+  onCloseSearch
+}) {
+  const [width, Height] = useWindowSize()
+  const [refSearchBar, serRefSearchBar] = React.useState({})
```
Comment (arajput) [inline packages/common-lib/src/components/SearchLayout.js:18]: correct the method name "serRefSearchBar"

File: packages/common-lib/src/index.tsx [hunk 2]
```diff
@@ -28,6 +29,7 @@ import * as teacherRegistryService from './services/teacherRegistryService'
 import * as classRegistryService from './services/classRegistryService'
```
Comment (arajput) [inline packages/common-lib/src/index.tsx:29]: common lib is wrong place to add Services

File: packages/worksheet/src/pages/WorksheetTemplate.js [hunk 3]
```diff
@@ -9,7 +9,7 @@ import { Box, Button, HStack, Skeleton, Stack, VStack } from "native-base";
 import React from "react";
 import { useTranslation } from "react-i18next";
 import manifest from "../manifest.json";
-import "react-responsive-carousel/lib/styles/carousel.min.css"; // requires a loader
+import "react-responsive-carousel/lib/styles/carousel.min.css";
```
Comment (arajput) [inline packages/worksheet/src/pages/WorksheetTemplate.js:12]: why css loaded here? can we move this to build script?

File: packages/common-lib/src/index.tsx (hunk 2 above)
Comment (sagarkoshti1990) [reply packages/common-lib/src/index.tsx:29]: can you suggest where to add this ?

File: packages/worksheet/src/pages/Worksheet.js [hunk 4]
```diff
@@ -67,45 +67,78 @@ const sortArray = [
...
   }
 
+  if (searchState) {
+    return (
+      <SearchLayout
+        {...{
+          search,
+          setSearch,
+          minStringLenght: 3,
+          notFoundMessage: "Type to Start Searching Worksheets",
```
Comment (arajput) [inline packages/worksheet/src/pages/Worksheet.js:131]: translation  required?

File: packages/worksheet/src/pages/CreateWorksheet.js [hunk 5]
```diff
@@ -112,13 +112,15 @@ export default function CreateWorksheet({ footerLinks, appName }) {
...
-          ? formObject.name
-            ? t("Your worksheet has been created.")
-            : t("You can see all questions here")
-          : pageName === "AddDescriptionPage"
-          ? t("Enter Worksheet Details")
-          : t("Show questions based on")
+        <H2 textTransform="inherit">
+          {pageName === "ListOfQuestions"
+            ? formObject.name
+              ? t("Your worksheet has been created.")
```
Comment (arajput) [inline packages/worksheet/src/pages/CreateWorksheet.js:118]: is it  correct convention for translation keys?

**Pull request merged**